// vuepress-theme-hope/lib/client/modules/info/components/CategoryInfo.js
import { defineComponent, h } from "vue";
import { generateIndexFromHash } from "vuepress-shared/client";
import { useNavigate } from "@theme-hope/composables/index";
import { CategoryIcon } from "@theme-hope/modules/info/components/icons";
import { useMetaLocale } from "@theme-hope/modules/info/composables/index";
import "../styles/category-info.scss";
export default defineComponent({
    name: "CategoryInfo",
    inheritAttrs: false,
    props: {
        /**
         * Category information
         *
         * åˆ†ç±»ä¿¡æ¯
         */
        category: {
            type: Array,
            required: true,
        },
        /**
         * Whether in pure mode
         *
         * æ˜¯å¦å¤„äºŽçº¯å‡€æ¨¡å¼
         */
        pure: Boolean,
    },
    setup(props) {
        const metaLocale = useMetaLocale();
        const navigate = useNavigate();
        return () => props.category.length
            ? h("span", {
                class: "page-category-info",
                "aria-label": `${metaLocale.value.category}${props.pure ? "" : "ðŸŒˆ"}`,
                ...(props.pure ? {} : { "data-balloon-pos": "up" }),
            }, [
                h(CategoryIcon),
                props.category.map(({ name, path }) => h("span", {
                    class: [
                        "page-category-item",
                        {
                            // TODO: magic number 9 is tricky here
                            [`category${generateIndexFromHash(name, 9)}`]: !props.pure,
                            clickable: path,
                        },
                    ],
                    role: path ? "navigation" : "",
                    onClick: () => {
                        if (path)
                            navigate(path);
                    },
                }, name)),
                h("meta", {
                    property: "articleSection",
                    content: props.category.map(({ name }) => name).join(","),
                }),
            ])
            : null;
    },
});
