// vuepress-theme-hope/lib/client/modules/info/components/TagInfo.js
import { defineComponent, h } from "vue";
import { generateIndexFromHash } from "vuepress-shared/client";
import { useNavigate } from "@theme-hope/composables/index";
import { TagIcon } from "@theme-hope/modules/info/components/icons";
import { useMetaLocale } from "@theme-hope/modules/info/composables/index";
import "../styles/tag-info.scss";
export default defineComponent({
    name: "TagInfo",
    inheritAttrs: false,
    props: {
        /**
         * Tag information
         *
         * æ ‡ç­¾ä¿¡æ¯
         */
        tag: {
            type: Array,
            default: () => [],
        },
        /**
         * Whether in pure mode
         *
         * æ˜¯å¦å¤„äºŽçº¯å‡€æ¨¡å¼
         */
        pure: Boolean,
    },
    setup(props) {
        const metaLocale = useMetaLocale();
        const navigate = useNavigate();
        return () => props.tag.length
            ? h("span", {
                class: "page-tag-info",
                "aria-label": `${metaLocale.value.tag}${props.pure ? "" : "ðŸ·"}`,
                ...(props.pure ? {} : { "data-balloon-pos": "up" }),
            }, [
                h(TagIcon),
                props.tag.map(({ name, path }) => h("span", {
                    class: [
                        "page-tag-item",
                        {
                            // TODO: magic number 9 is tricky here
                            [`tag${generateIndexFromHash(name, 9)}`]: !props.pure,
                            clickable: path,
                        },
                    ],
                    role: path ? "navigation" : "",
                    onClick: () => {
                        if (path)
                            navigate(path);
                    },
                }, name)),
                h("meta", {
                    property: "keywords",
                    content: props.tag.map(({ name }) => name).join(","),
                }),
            ])
            : null;
    },
});
